swarm/storage/mru/topic.go

```diff
@@ -0,0 +1,56 @@
+package mru
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/ethereum/go-ethereum/common/bitutil"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/swarm/storage"
+)
+
+const topicLength = storage.KeyLength
+
+type Topic struct {
+	content [topicLength]byte
+}
+
+func NewTopic(name string, relatedContent storage.Address) Topic {
+	var topic Topic
+	copy(topic.content[:], relatedContent)
+	nameLength := len(name)
+	if nameLength > topicLength {
+		nameLength = topicLength
+	}
+	bitutil.XORBytes(topic.content[:], topic.content[:], []byte(name)[:nameLength])
+	return topic
+}
+
+func (t *Topic) Hex() string {
+	return hexutil.Encode(t.content[:])
+}
+
+func (t *Topic) FromHex(hex string) error {
+	return decodeHexArray(t.content[:], hex, "Topic")
+}
+
+func (t *Topic) Name(relatedContent storage.Address) string {
+	nameBytes := t
+	bitutil.XORBytes(nameBytes.content[:], t.content[:], relatedContent)
```

Comment:
Why new assignment to overwrite?

Reply:
I am making a copy because the XOR after that is destructive, and I don't want the topic to be affected by extracting out the name